### ISSUE TITLE
Confusion Matrix

### ISSUE THREAD

**Issue by zbpvarun** [NickleDave/hybrid-vocal-classifier#55]
Adding confusion matrix would help with select.

import hvc
ls
cd pu63pu64
hvc.extract('gy6or6_autolabel.example.extract.svm.config.yml')
pwd
hvc.extract('gy6or6_autolabel.example.extract.svm.config.yml')
ls
cd extract_output_180125_112057/
from sklearn.externals import joblib
ftrs = hvc.utils.load_feature_file('summary_feature_file_created_180125_112115')
ftrs.keys()
ftr['features'].shape
ftrs['features'].shape
hvc.extract('gy6or6_autolabel.example.extract.svm.config.yml')
cd ..
hvc.extract('gy6or6_autolabel.example.extract.svm.config.yml')
hvc.extract('gy6or6_autolabel.example.extract.svm.config.yml')
pwd
ls
cd extract_output_180125_112706/
ftrs = hvc.utils.load_feature_file('summary_feature_file_created_180125_112732')
ftrs['features'].shape
hvc.utils.grid_search_svm_rbf?
best_params, best_score, cv_results = hvc.utils.grid_search_svm_rbf(ftrs['features'],ftrs['labels'])
whos
best_params, best_score = hvc.utils.grid_search_svm_rbf(ftrs['features'],ftrs['labels'])
ls
cd ..
ls
hvc.select('pu63pu64_autolabel.example.select.svm.config.yml')
hvc.select('pu63pu64_autolabel.example.select.svm.config.yml')
from sklearn.metrics import confusion_matrix as confmat
ls
cd pu63pu64_autolabel_test/
ls
cd select_output_180125_120226/
ls
cd svm_C100.0_gamma1e-06/
ls
whos
samp950_rep9 = joblib.load('svm_950samples_replicate9.meta')
samp950_rep9.keys()
cd ..
ls
select_summ = joblib.load('summary_model_select_file_created_180125_120226')
select_summ.keys()
whos
confmat?
ls
select_summ.keys()
test_IDs = select_summ['test_IDs']
select_summ['train_IDs_arr'].shape
train_IDs = select_summ['train_IDs_arr'][19,9]
labels = ftrs['labels']
labels.shape
len(labels)
pred_labels = select_summ['pred_labels_arr']
pred_labels[19,9]
labels = np.asarray(labels)
import numpy as np
labels = np.asarray(labels)
labels
labels[train_IDs[19,9]]
train_IDs[19,9]
train_IDs
labels[train_IDs]
y_pred = labels[train_IDs]
whos
y_true = labels[test_IDs]
y_pred.shape
y_true.shape
y_pred = pred_labels[19,9]
y_true = labels[train_IDs]
confmat?
confmat(y_true, y_pred)
y_true.shape
y_pred.shape
whso
wyhos
whos
pred_labels[19,9]
y_pred = pred_labels[19,9]
y_pred.shape
y_pred = pred_labels[19,9][0]
y_pred.shape
y_true = labels[test_IDs]
y_true.shape
y_true[:20]
y_pred[:20]
confmat(y_true,y_pred)
import matplotlib.pyplot as plt
cm = confmat(y_true,y_pred)
labelset = ftrs['labelset']
labelset
confmat?
cm = confmat(y_true,y_pred, labelset)
plt.colormesh(labelset, labelset, cm)
plt.pcolormesh(labelset, labelset, cm)
labelset
plt.pcolormesh(cm)
plt.show()
fig, ax = plt.subplots()
ax.pcolormesh(cm)
ax.set_xticklabels(labelset)
ax.set_yticklabels(labelset)
fig.show()
ax.imshow(cm)
ax.set_xticks(range(labelset.shape[-1]))
ax.set_xticks(range(len(labelset)))
ax.set_yticks(range(len(labelset)))
ax.set_xticklabels(labelset)
ax.set_yticklabels(labelset)
plt.show()
fig.show()
fig, ax = plt.subplots()
ax.imshow(cm)
ax.set_xticklabels(labelset)
ax.set_yticks(range(len(labelset)))
ax.set_xticks(range(len(labelset)))
ax.set_xticklabels(labelset)
ax.set_yticklabels(labelset)
plt.show()
confmat?
cm_norm = cm / cm.sum(axis=1)
cm_norm
plt.imshow(cm_norm)
plt.show()
plt.imshow(cm_norm)
plt.colorbar()
plt.show()
cm_norm2 = cm.astype(float) / cm.sum(axis=1)[:, np.newaxis]
np.array_equal(cm_norm, cm_norm2)
cm_norm = cm_norm2
plt.imshow(cm_norm)
plt.colorbar()
plt.xticks(range(len(labelset)))
plt.yticks(range(len(labelset)));
plt.xticklabels(labelset);
plt.xticks?
plt.xticks(range(len(labelset)), labelset)
plt.yticks(range(len(labelset)), labelset);
plt.show()
%save?
%save conf_mat.py
%save -r conf_mat.py
%script?
%history -f conf_mat.py

**Comment by NickleDave** [comment]
added in 134a9f3
